### DISCUSSION TITLE
Suggestion: Be able to pin articles.

### DISCUSSION THREAD

**Discussion by white-gorilla** [squidfunk/mkdocs-material#5047] [Q&A] [answered]
I have integrated the blog in standalone and [it is great](https://ai.lefebvre-sarrut.eu/).

What would be interesting would be to have the possibility to have a metadata, like the `draft`, to be able to pin some articles (let's imagine a `pin: true` meta).
The articles pinned in this way would be placed at the top of the index, regardless of their date of writing, with a special emphasis.

This is a very common feature after all.

**Comment by squidfunk** [comment]
That sounds like a good idea! We'll definitely consider it for future addition 😊a

**Reply by meysam81** [reply]
Any update on this? It's a really interesting feature.

**Reply by squidfunk** [reply]
Not yet, but we can definitely add it in the future. However, until now, it doesn't seem that many people need it.

**Reply by meysam81** [reply]
I'd be interested in contributing. Just need a bit of help of where to start with. :pray:

**Reply by squidfunk** [reply]
Thanks for offering! However, we first need to make sure that we actually _want_ this in our code base, because in the end, even though it might be externally contributed, _it is us maintaining it_. For this reason, we'll first let this sit here a little and see if other users might find it useful 😉

**Reply by kamilkrzyskow** [reply]
@meysam81 Perhaps you would like to implement it for yourself, and share your customization with others?
To add a `pinned` section for blog posts at the top of the blog index you have to override the `blog.html` template:
https://squidfunk.github.io/mkdocs-material/customization/#extending-the-theme

https://github.com/squidfunk/mkdocs-material/blob/23f18866c342968f5819495596fcb24f5dcc8b85/src/templates/blog.html#L30-L38

Between the header and posts you have to add the logic for the pinned pages, where you input the appropriate HTML structure. 
Apart of that additional CSS will be needed to make it look nice: https://squidfunk.github.io/mkdocs-material/customization/#additional-css

```jinja2
      <!-- Pins -->
      {% for post in posts %}
      
      {% if post.config.pin %}   
      {% include "partials/post-pinned.html" %}
      {% endif %}
      
      {% endfor %}
```

The `post.config` also includes the `post.meta`, so the `pin` key is looked up as per the OP.

**Reply by squidfunk** [reply]
Thanks for providing guides @kamilkrzyskow 

However, please note that this implementation will repeat pinned blog posts on paginated pages, which is very likely not what is expected by the user. Pinned block posts should always appear on the first page of the blog, but not on subsequent pages. Additionally, pinned posts will also appear on paginated pages of the archive and category pages.

**Comment by shenweiyan** [comment]
Great advice, I also need such a feature.

**Comment by squidfunk** [comment, marked as answer]
As this has been asked a few times, I decided to squeeze this in between the other features that I'm currently working on. [`93a2ccc`](https://github.com/squidfunk/mkdocs-material-insiders/commit/93a2ccca1c050c0ce73be3a442a2eefee492f71a) adds support for pinning of blog posts! Pinned blog posts will be sorted before all other blog posts. If you pin multiple blog posts, they are sorted by descending date, like regular blog posts. Pinned posts will appear on the first page of the main blog page, as well as on the first page of archive and category pages if and only if they belong to those pages, but not on paginated pages.

To pin a blog post, just use the `pin` property in the Markdown file of the blog post:

``` md
---
date: 2024-02-24
pin: true
---

# Post title
```

Here's how it looks – a small pin icon is rendered at the end of the metadata of the post + a horizontal line at the bottom to signal that this post is outside of the normal flow of documents:

![screenshot-localhost-3000-mkdocs-material-blog-1708749583337](https://github.com/squidfunk/mkdocs-material/assets/932156/c44e249f-6164-449b-96ef-6fb5c5ac174d)

**Reply by squidfunk** [reply]
Note that this feature is part of [Insiders](https://squidfunk.github.io/mkdocs-material/insiders/) and currently only available to sponsors.

**Reply by squidfunk** [reply]
Released as part of [9.5.11+insiders-4.53.0](https://squidfunk.github.io/mkdocs-material/insiders/changelog/#4.53.0).

**Reply by JieyinFeng** [reply]
Hello! I am glad to find this answer! I would like to know if this function is still under "sponsors"? I just tried it but it didn't show on my end. Please, any update on this function would be appreciated ~~~ Thank you!

**Reply by kamilkrzyskow** [reply]
@JieyinFeng It's still a sponsors only feature, and will stay like that until it reaches this sponsoring tier: https://squidfunk.github.io/mkdocs-material/insiders/benefits/#18000-lemon-drop

**Reply by JieyinFeng** [reply]
Thank you!
